### ISSUE TITLE
Allow function assignment using require-optimization

### ISSUE THREAD

**Issue by rdiazv** [yannickcr/eslint-plugin-react#779]
```es6
import React from 'react';
import shouldPureComponentUpdate from 'helpers/shouldPureComponentUpdate';

class Sample extends React.Component {
  shouldComponentUpdate = shouldPureComponentUpdate;

  render() {
    return null;
  }
}
```

This throws `Component is not optimized. Please add a shouldComponentUpdate method`.

**Comment by ljharb** [comment]
That makes it an instance property, not a prototype property, which isn't really what i'd call "optimization".

Do this instead:
```js
shouldComponentUpdate(...args) {
  return shouldPureComponentUpdate(...args);
}
```

**Comment by rdiazv** [comment]
Nice, thanks! 👍
